qcodes/tests/test_station.py

```diff
@@ -201,3 +215,306 @@ def test_station_after_instrument_is_closed():
+
+
+@pytest.fixture
+def example_station(example_station_config):
+    return Station(config_file=example_station_config)
+
+
+# instrument loading related tests
+def test_station_config_path_resolution(example_station_config):
+    config = qcodes.config["station_configurator"]
+
+    assert not has_station_config_been_loaded(Station())
+
+    path = Path(example_station_config)
+    config["default_file"] = str(path)
+    assert has_station_config_been_loaded(Station())
+
+    config["default_file"] = path.name
+    config["default_folder"] = str(path.parent)
+    assert has_station_config_been_loaded(Station())
+
+    config["default_file"] = 'random.yml'
+    config["default_folder"] = str(path.parent)
+    assert not has_station_config_been_loaded(Station())
+
+    config["default_file"] = str(path)
+    config["default_folder"] = r'C:\SomeOtherFolder'
+    assert has_station_config_been_loaded(Station())
+
+    config["default_file"] = None
+    config["default_folder"] = str(path.parent)
+    assert has_station_config_been_loaded(Station(config_file=path.name))
+
+    config["default_file"] = None
+    config["default_folder"] = None
+    assert has_station_config_been_loaded(Station(config_file=str(path)))
+
+
+def test_station_creation(example_station):
+    assert "StationConfigurator" in example_station.components.keys()
+
+SIMPLE_MOCK_CONFIG = """
+instruments:
+  mock:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+"""
+
+@pytest.fixture
+def simple_mock_station(example_station_config):
+    yield station_from_config_str(SIMPLE_MOCK_CONFIG)
+
+def test_simple_mock_config(simple_mock_station):
+    st = simple_mock_station
+    assert has_station_config_been_loaded(st)
+    assert hasattr(st, 'load_mock')
+    mock_snapshot = st.snapshot()['components']['StationConfigurator']\
+        ['instruments']['mock']
+    assert mock_snapshot['driver'] == "qcodes.tests.instrument_mocks"
+    assert mock_snapshot['type'] == "DummyInstrument"
+
+
+def test_simple_mock_load_mock(simple_mock_station):
+    st = simple_mock_station
+    mock = st.load_mock()
+    assert isinstance(mock, DummyInstrument)
+    assert mock.name == 'mock'
+    assert st.components['mock'] is mock
+
+
+def test_simple_mock_load_instrument(simple_mock_station):
+    st = simple_mock_station
+    mock = st.load_instrument('mock')
+    assert isinstance(mock, DummyInstrument)
+    assert mock.name == 'mock'
+    assert st.components['mock'] is mock
+
+
+def test_enable_force_reconnect() -> None:
+    def get_instrument_config(enable_forced_reconnect: Union[bool, None]) -> str:
+        return f"""
+instruments:
+  mock:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+    {f'enable_forced_reconnect: {enable_forced_reconnect}'
+        if enable_forced_reconnect is not None else ''}
+    init:
+      gates: {{"ch1", "ch2"}}
+         """
+
+    def assert_on_reconnect(user_config_val: Union[bool, None],
```

Comment:
I have used `Union` here on purpose because `None` actually acts as a value more than as a default. Let us try to find out what the intended use of `Optional` is.

Reply:
From the typing docs `Optional[X] is equivalent to Union[X, None].` so I think those are fully equivalent. https://docs.python.org/3/library/typing.html#typing.Optional